src/main/frontend/src/components/AddCatalogPriceForm.js

```diff
@@ -2,12 +2,166 @@
 // IMPORTANT:
 // You must update ResourceUrl.RESOURCES_VERSION each time whenever you're modified this file!
 //
+// @todo #1342 AddCatalogPriceForm: show a currency
 
 class AddCatalogPriceForm extends React.Component {
-	
+	constructor(props) {
+		super(props);
+		this.state = {
+			price: null,
+			catalog: '',
+			hasServerError: false,
+			validationErrors: []
+		};
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.handleChangePrice = this.handleChangePrice.bind(this);
+		this.handleChangeCatalog = this.handleChangeCatalog.bind(this);
+	}
+
+	handleChangePrice(event) {
+		event.preventDefault();
+		this.setState({
+			price: event.target.value
+		});
+	}
+
+	handleChangeCatalog(event) {
+		event.preventDefault();
+		this.setState({
+			catalog: event.target.value
+		});
+	}
+
+	handleSubmit(event) {
+		event.preventDefault();
+
+		this.setState({
+			isDisabled: true,
+			hasServerError: false,
+			validationErrors: []
+		});
+
+		axios
+			.patch(
+				this.props.url,
+				[
+					{
+						op: 'add',
+						path: `/${this.state.catalog}_price`,
+						value: this.state.price
+					}
+				],
+
+				{
+					headers: {
+						[this.props.csrfHeaderName]: this.props.csrfTokenValue,
+						'Cache-Control': 'no-store'
+					},
+					validateStatus: status => {
+						return status == 204 || status == 400;
+					}
+				}
+			)
+			.then(response => {
+				const data = response.data;
+				if (data.hasOwnProperty('fieldErrors')) {
+					const fieldErrors = [];
+					if (data.fieldErrors.price) {
+						fieldErrors.push(...data.fieldErrors.price);
+					}
+
+					this.setState({
+						isDisabled: false,
+						validationErrors: fieldErrors
+					});
+					return;
+				}
+
+				// no need to reset the state as page will be reloaded
+				window.location.reload();
+			})
+			.catch(error => {
+				console.error(error);
+				this.setState({ isDisabled: false, hasServerError: true });
+			});
+	}
 	render() {
+		const hasValidationErrors = this.state.validationErrors.length > 0;
 		return (
-			<div></div>
-		)
+			<div className="col-sm-12 form-group">
+				<form className="form-horizontal" onSubmit={this.handleSubmit}>
```

Comment:
In order to make an errored field to be highlighted properly we also should have `has-error` class on a form:
```suggestion
				<form className={`form-horizontal ${hasValidationErrors ? 'has-error' : ''}`} onSubmit={this.handleSubmit}>
```